core/src/main/java/com/emc/ia/sdk/support/http/HttpClient.java

```diff
@@ -29,7 +29,7 @@
     return post(uri, headers, type, payload);
   }
 
-  <T> T post(String uri, Collection<Header> headers, Class<T> type, Part... parts) throws IOException;
+  <T> T post(String uri, Collection<Header> headers, Class<T> type, Collection<Part> parts) throws IOException;
```

Comment:
One way would be to fix it in the caller: `new Part[] { part }`. The other way is to fix it in the method: check whether `payload instanceof Part`. I prefer the latter.